### DISCUSSION TITLE
Feedback on custom domain binding issue

### DISCUSSION THREAD

**Discussion by jiuyuesanqiu** [iterate-ch/cyberduck#14575] [Q&A]
Hello everyone,

I have encountered an issue when connecting to my Aliyun storage bucket using the Mountain Duck software. Here are the details of my situation:

I am currently using the oss-cn-shenzhen.aliyuncs.com server in Shenzhen, and I am connecting to a storage bucket called my-bucket using the S3 protocol. In the connection editor, I have set the path to /my-bucket. I have also bound a custom domain my-custom-domain.com to my storage bucket.

While I am able to upload and download files, the URL that is copied when I right-click a file is incorrect. For example, when I copy the URL of the file a.png, the URL that is copied is http://oss-cn-shenzhen.aliyuncs.com/my-bucket/a.png. However, the actual URL should be https://my-bucket.oss-cn-shenzhen.aliyuncs.com/a.png or https://my-custom-domain.com/a.png.

I suspect that this issue is caused by Mountain Duck's inability to correctly resolve the custom domain. I am using the latest version of Mountain Duck, and I have also tried manually entering the correct URL, but the issue persists.

Could you please advise me on any possible methods to enable Mountain Duck to correctly resolve my custom domain and copy the correct URL? I would appreciate any assistance or guidance that you can provide.

Thank you for your help and support.

**Comment by AliveDevil** [comment]
This feature isn't exposed in Mountain Duck directly.

Workaround is to install Cyberduck, then edit that connection, expand the Extended-properties and enter `"https://my-custom-domain.tld"` into the [WebURL](https://docs.cyberduck.io/cyberduck/bookmarks/#http-url)-field, this is then recognized by Mountain Duck and should return the correct URL for your files.

**Reply by jiuyuesanqiu** [reply]
Thank you for providing a workaround for the custom domain binding issue. However, after following the steps provided and adding my custom domain to the WebURL field in Cyberduck, I still see two URLs when I copy the file URL. The URLs copied are in the following format: https://oss-cn-shenzhen.aliyuncs.com/my-bucket/a.png and https://my-custom-domain.com/a.png.

I have confirmed that I have correctly added the custom domain to the WebURL field and have edited the connection in Mountain Duck to be associated with the one in Cyberduck.

Please let me know if there is anything else that I can try to resolve this issue.

Thank you for your help and support.

**Reply by AliveDevil** [reply]
So, the Mountain Duck context menu copies both URLs, instead of separating them, like Cyberduck does. Will discuss this.

Cyberduck, for Copy URL:
- HTTP URL
`http://bucket.service.com/path`
- HTTP URL
`http://custom-domain.tld/path`

Which allows copying each entry individually, while on Mountain Duck it is
- HTTP URL
`http://bucket.service.com/path`
`http://custom-domain.tld/path`

Which copies both entries into the clipboard.
